raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -153,6 +162,25 @@ def has_secret_registration_started(
     return is_secret_registered_onchain or has_pending_transaction
 
 
+def filter_reachable_routes(
+        routes: typing.List[RouteState],
+        nodeaddresses_to_networkstates: typing.NodeNetworkStateMap,
+):
+    """This function makes sure we use reachable routes only."""
+    address_to_route = {r.node_address: r for r in routes}
+
+    for route in routes:
+        node_network_state = nodeaddresses_to_networkstates.get(
+            route.node_address,
+            NODE_NETWORK_UNREACHABLE,
+        )
+
+        if node_network_state != NODE_NETWORK_REACHABLE:
+            del address_to_route[route.node_address]
+
+    return list(address_to_route.values())
```

Comment:
This is changing the order of the elements, meaning that the ranking done by the PFS is lost. Please fix this to preserve the order. (This is also bad behavior with the existing code `filter_used_routes`)

Additionally, if you can please add a comment about this in the docstring and maybe a test?